daemon/containerd/image_pull.go

```diff
@@ -48,17 +49,40 @@ func (i *ImageService) PullImage(ctx context.Context, image, tagOrDigest string,
 	resolver, _ := i.newResolverFromAuthConfig(ctx, authConfig)
 	opts = append(opts, containerd.WithResolver(resolver))
 
+	old, err := i.resolveDescriptor(ctx, ref.String())
+	if err != nil && !errdefs.IsNotFound(err) {
+		return err
+	}
+	p := platforms.Default()
+	if platform != nil {
+		p = platforms.Only(*platform)
+	}
+
+	available, _, _, missing, err := images.Check(ctx, i.client.ContentStore(), old, p)
+	if err != nil {
+		return err
+	}
+
 	jobs := newJobs()
 	h := images.HandlerFunc(func(ctx context.Context, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
-		if desc.MediaType != images.MediaTypeDockerSchema1Manifest {
+		if images.IsLayerType(desc.MediaType) {
 			jobs.Add(desc)
 		}
 		return nil, nil
 	})
 	opts = append(opts, containerd.WithImageHandler(h))
 
 	out := streamformatter.NewJSONProgressOutput(outStream, false)
-	finishProgress := jobs.showProgress(ctx, out, pullProgress{Store: i.client.ContentStore(), ShowExists: true})
+	var pp progressUpdater
+
+	// If we already have all the contents pull shouldn't show any layer
```

Comment:
What will this show when the image is updated and new layers are pulled?

Reply:
The wrong thing… okay I’ll have to go back to the initial image handler